wrapper/CSharp/wolfSSL_CSharp/wolfSSL.cs

```diff
@@ -569,13 +574,35 @@ public static int read(IntPtr ssl, StringBuilder buf, int sz)
             try
             {
                 IntPtr sslCtx = unwrap(ssl);
+                IntPtr data;
+                int ret;
+                byte[] msg;
+
                 if (sslCtx == IntPtr.Zero)
                 {
-                    log(ERROR_LOG, "connect error");
+                    log(ERROR_LOG, "read error");
                     return FAILURE;
                 }
+                data = Marshal.AllocHGlobal(sz);
+
+                ret = wolfSSL_read(sslCtx, data, sz);
+
+                if (ret >= 0)
+                {
+                    /* Get data that was sent accross and store it using a literal read of
+                     * the conversion from bytes to character. Takes care of if
+                     * a null terminator is part of the message read.
+                     */
+                    msg = new byte[ret];
+                    Marshal.Copy(data, msg, 0, ret);
+                    for (int i = 0; i < ret; i++)
+                    {
+                        buf.Append(@Convert.ToChar(msg[i]));
```

Comment:
Hey Jacob, worried about the Convert.ToChar(byte) handling non-printable data correctly. Manual says "Converts the value of the specified 8-bit unsigned integer to its equivalent Unicode character". Is this "read" version of the API meant for just string data and if reading binary data would use the other byte[] version of the "read" API below?